pocket.go

```diff
@@ -12,125 +12,127 @@ import (
-		os.Exit(1)
-	}
-
-	argument := os.Args[1]
-	if argument == "-h" || argument == "--help" || argument == "help" || argument == "-help" {
-		fmt.Printf("Usage: %s [get|add|forget]\n", os.Args[0])
-		os.Exit(2)
-	} else {
-		mode = argument
-	}
-
-	// Verify that mode is valid.
-	if mode != "get" && mode != "add" && mode != "forget" {
-		fmt.Println("[!] invalid mode; use `pocket help`")
-		os.Exit(1)
+	mode, err := auxiliary.ParseArgs(os.Args)
+	if err != nil {
+		if err.Error() == "help" {
+			os.Exit(0)
+		} else {
+			fmt.Println(err)
+			os.Exit(1)
+		}
 	}
 
 	// Run setup.
 	auxiliary.Setup()
 
-	// Prompt user for the password without echoing back.
-	password := auxiliary.GetPass("[-] password: ")
-	if len(password) < 1 {
-		fmt.Println("[!] length of password must be non-zero")
-		os.Exit(1)
-	}
-
 	// Prompt user for identifier.
-	id := []byte(auxiliary.Input("[-] identifier: "))
-	if len(id) < 1 {
-		fmt.Println("[!] length of identifier must be non-zero")
+	identifier = []byte(auxiliary.Input("[-] Identifier: "))
+	if len(identifier) < 1 {
+		fmt.Println("[!] Length of identifier must be non-zero.")
 		os.Exit(1)
 	}
 
-	// Derive and store encryption key.
-	fmt.Println("[+] deriving encryption key...")
-	key = crypto.DeriveKey(password, id)
-
-	// Derive and store identifier.
-	fmt.Println("[+] deriving secure identifier...")
-	identifier = crypto.DeriveID(id)
-
 	// Grab pre-saved secrets.
 	secretData = auxiliary.RetrieveSecrets()
 
 	// Launch appropriate function for run-mode.
 	switch mode {
-	case "get":
-		retrieve()
-	case "add":
-		add()
-	case "forget":
+	case "get": // id pass
+		// Prompt user for the password without echoing back.
+		password := auxiliary.GetPass("[-] Password: ")
+		if len(password) < 1 {
+			fmt.Println("[!] Length of password must be non-zero.")
+			os.Exit(1)
+		}
+		retrieve(password)
+	case "add": // id pass secret
+		// Prompt user for the password without echoing back.
+		password := auxiliary.GetPass("[-] Password: ")
+		if len(password) < 1 {
+			fmt.Println("[!] Length of password must be non-zero.")
+			os.Exit(1)
+		}
+		add(password)
+	case "forget": // id
 		forget()
 	}
 
-	// Clear sensitive data from memory. (Probably not secure, but good enough.)
-	key = []byte("")
-	identifier = ""
+	// Clear sensitive data from memory.
+	// (Probably not secure, but good enough.)
+	identifier = []byte("")
 }
 
-func retrieve() {
-	secret := secretData[identifier]
+func retrieve(password []byte) {
+	// Derive and store identifier.
+	fmt.Println("[+] Deriving secure identifier...")
+	id := crypto.DeriveID(identifier)
+
+	// Derive and store encryption key.
+	fmt.Println("[+] Deriving encryption key...")
+	key := crypto.DeriveKey(password, identifier)
+
+	secret := secretData[id]
 	if secret != nil {
 		secret := crypto.Unpad(crypto.Decrypt(secret.(string), key))
-		fmt.Println("[+] secret:", string(secret))
+		fmt.Println("[+] Secret:", string(secret))
 	} else {
```

Comment:
The use of else is kind of frowned upon in Go. When you try to get the value from a map, it returns a second `bool`, which is whether it was found. I recommend changing this to:
```
secret, ok := secretData[id]
if !ok || secret == nil {
    // handle what happens when the secret doesn't exist
}

// continue with the program flow
```